src/gar.lua

```diff
@@ -2055,7 +2067,6 @@ function unlockGatewayDelegateVault(gateway, delegateAddress, vaultId)
 	assert(vault, "Vault not found")
 
 	balances.increaseBalance(delegateAddress, vault.balance)
-	gateway.delegates[delegateAddress] = nil
```

Comment:
we nil out delegate in `decreaseDelegateStakeAtGateway` if necessary

Reply:
i'm also not sure if we should be calling `decreaseDelegateStakeAtGateway` here

Reply:
[7b55fd5](https://github.com/ar-io/ar-io-network-process/pull/283/commits/7b55fd5ee23add7948d21d23b07ccc393745d35c)